timesketch/api/v1/resources/archive.py

```diff
@@ -424,96 +442,364 @@ def _export_view(self, view, sketch, zip_file):
+                f"Failed to archive sketch. Details: {'; '.join(error_details)}",
+            )
 
-        return HTTP_STATUS_CODE_OK
+        # 4. If all indices are closed, update the database statuses.
+        sketch.set_status(status="archived")
+        for timeline in sketch.timelines:
+            timeline.set_status(status="archived")
+            logger.debug(
+                "Sketch %s Timeline %s status set to 'archived'.",
+                sketch.id,
+                timeline.id,
+            )
+        for search_index in successfully_closed_indexes:
+            search_index.set_status(status="archived")
+            logger.debug(
+                "Sketch %s search_index %s status set to 'archived'.",
+                sketch.id,
+                search_index.id,
+            )
+        for search_index in failed_to_find_indexes:
+            search_index.set_status(status="fail")
+            logger.debug(
+                "Sketch %s search_index %s status set to 'fail'.",
+                sketch.id,
+                search_index.id,
+            )
+        logger.info("Sketch %s status set to 'archived'.", sketch.id)
 
-    def _archive_sketch(self, sketch: Sketch):
-        """Unarchives a sketch by opening up all indices and removing labels.
+        # Commit changes after processing all indices
+        db_session.commit()
+
+        return jsonify({"message": f"Sketch {sketch.id} has been archived."})
+
+    def _unarchive_sketch(self, sketch: Sketch):
+        """Unarchives a sketch, making it and its data active again.
+
+        This method follows a transactional approach to ensure data consistency.
+        It first attempts to open all necessary OpenSearch indices associated
+        with the sketch's archived timelines. If any index fails to open for a
+        critical reason (e.g., not found), the entire operation is aborted,
+        leaving the sketch in its archived state.
+
+        If all indices are successfully opened (or confirmed to be already open),
+        the method proceeds to update the database, setting the status of the
+        sketch, its timelines, and the corresponding SearchIndex objects to 'ready'.
 
         Args:
-            sketch: Instance of timesketch.models.sketch.Sketch
+            sketch: An instance of timesketch.models.sketch.Sketch to unarchive.
+
+        Returns:
+            An integer representing the HTTP status of the operation.
+            - HTTP_STATUS_CODE_OK if successful.
+
+        Raises:
+            HTTPException:
+                - If the sketch is not currently archived.
+                - If any timeline within the sketch is in a state that prevents
+                  unarchiving (e.g., 'processing').
+                - If a critical error occurs while opening an OpenSearch index.
         """
-        if sketch.get_status.status == "archived":
+        logger.info("Unarchiving sketch %s ('%s').", sketch.id, sketch.name)
```

Comment:
Debatable: Should info logging contain potentially sensitive case data like `sketch.name`? We do try to avoid it for error and warnings. But info is something turned on intentionally by the admins?